docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,492 @@
+      * **config**: json map that is mapped to configuration provided in the type.
+    * **endpoints**: defines a list of endpoints
+      * **address**: defines an address to which a user want to send a request. Is possible to provide `domain`, `ip` and `unix` sockets
+      * **port**: defines a port of a destination.
+    * **tls**: provides a TLS configuration when proxy is resposible for a TLS origination
+      * **version**: section for providing version specification.
+        * **min**: defines minmum supported version. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, `TLSv1_3`
+        * **max**: defines maximum supported version. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, `TLSv1_3`
+      * **allowRenegotiation**: defines if TLS sessions will allow renegotiation.
+      * **verification**: section for providing TLS verification details.
+        * **skipSAN**: defines if proxy should skip SAN verification. Default `false`.
+        * **subjectAltNames**: list of names to verify in the certificate.
+        * **caCert**: defines a certificate of CA.
+          * one of `inline`, `inlineString` or `secret`.
+        * **clientCert**: defines a certificate of a client.
+          * one of `inline`, `inlineString` or `secret`.
+        * **clientKey**: defines a client private key.
+          * one of `inline`, `inlineString` or `secret`.
+* **status**: status of an object managed by Kuma control-plane
+  * **vip**: section for allocated IP
+    * **value**: allocated IP for a provided domain with `KumaHostname` type in a match section or provided IP
+    * **type**: provides information about the way IP was provided
+  * **addresses**: section for generated domain
+    * **hostname**: generated domain 
+    * **status**: indicate if an address is available
+    * **origin**: section providing information what generated the vip
+      * **kind**: points to kind that generated domain
+      * **name**: name of the `HostnameGenerator`
+    * **reason**: holds error messages if there are any 
+
+#### Default CA certificate
+
+By default, when TLS is enabled and CA certificate is not set, we are going to use system bundled one. We can achive that by scanning in `kuma-dp` [default paths](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/security/ssl#example-configuration) for each environment and later send that information in the bootstrap request to the control-plane for a configuration generation.
+
+#### Policy matching
+
+`MeshExternalServices` should work the same with policies as `MeshService`. We are going to introduce a new `kind: MeshExternalService`, which allows targeting them with policies.
+
+#### Envoy resources naming
+
+Currently, we name clusters after their respective policies. Since the model includes only one match we can stick to the same pattern but prefix it with `mes_` which stands for `MeshExternalService`.
```

Comment:
`mes_` might be confusing. When someone don't know or remember that `mes` is an abbreviation from `MeshExternalService`, they might think it's just the name of policy name or maybe it's `mesh`? I don't think we are restricted by the length in the case of cluster names, so maybe we could just name it `meshexternalservice_httpbin`, or maybe even `meshexternalservice:httpbin` with `alt_stat_name: meshexternalservice_httpbin`?

Reply:
there is no mention of length limit in cluster docs https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/cluster/v3/cluster.proto